components/kubeflow/kfserving/src/kfservingdeployer.py

```diff
@@ -34,103 +34,231 @@
-        args=custom_model_spec.get('args', None),
-        image_pull_policy=custom_model_spec.get('image_pull_policy', None),
-        working_dir=custom_model_spec.get('working_dir', None)
+        command=custom_model_spec.get("command", None),
+        args=custom_model_spec.get("args", None),
+        image_pull_policy=custom_model_spec.get("image_pull_policy", None),
+        working_dir=custom_model_spec.get("working_dir", None),
+    )
+    return V1alpha2EndpointSpec(
+        predictor=V1alpha2PredictorSpec(
+            custom=V1alpha2CustomSpec(container=containerSpec)
+        )
     )
-    return V1alpha2EndpointSpec(predictor=V1alpha2PredictorSpec(custom=V1alpha2CustomSpec(container=containerSpec)))
 
 
-def InferenceService(metadata, default_model_spec, canary_model_spec=None, canary_model_traffic=None):
-    return V1alpha2InferenceService(api_version=constants.KFSERVING_GROUP + '/' + constants.KFSERVING_VERSION,
-                             kind=constants.KFSERVING_KIND,
-                             metadata=metadata,
-                             spec=V1alpha2InferenceServiceSpec(default=default_model_spec,
-                                                               canary=canary_model_spec,
-                                                               canary_traffic_percent=canary_model_traffic))
+def InferenceService(
+    metadata, default_model_spec, canary_model_spec=None, canary_model_traffic=None
+):
+    return V1alpha2InferenceService(
+        api_version=constants.KFSERVING_GROUP + "/" + constants.KFSERVING_VERSION,
+        kind=constants.KFSERVING_KIND,
+        metadata=metadata,
+        spec=V1alpha2InferenceServiceSpec(
+            default=default_model_spec,
+            canary=canary_model_spec,
+            canary_traffic_percent=canary_model_traffic,
+        ),
+    )
 
 
-def deploy_model(action, model_name, default_model_uri, canary_model_uri, canary_model_traffic, namespace, framework, default_custom_model_spec, canary_custom_model_spec, autoscaling_target=0):
+def deploy_model(
+    action,
+    model_name,
+    default_model_uri,
+    canary_model_uri,
+    canary_model_traffic,
+    namespace,
+    framework,
+    default_custom_model_spec,
+    canary_custom_model_spec,
+    service_account,
+    autoscaling_target=0,
+):
     if int(autoscaling_target) != 0:
         annotations = {"autoscaling.knative.dev/target": str(autoscaling_target)}
     else:
         annotations = None
-    metadata = client.V1ObjectMeta(name=model_name, namespace=namespace, annotations=annotations)
-    
+    metadata = client.V1ObjectMeta(
+        name=model_name, namespace=namespace, annotations=annotations
+    )
+
     # Create Default deployment if default model uri is provided.
-    if framework != 'custom' and default_model_uri:
+    if framework != "custom" and default_model_uri:
         default_model_spec = EndpointSpec(framework, default_model_uri)
```

Comment:
Maybe you need to pass the `service_account` here in order to assign it to the Endpoint Spec